src/inference/src/threading/ie_executor_manager.cpp

```diff
@@ -90,9 +131,55 @@ void ExecutorManagerImpl::clear(const std::string& id) {
     }
 }
 
-ExecutorManager::Ptr executorManager() {
-    static auto manager = std::make_shared<ExecutorManagerImpl>();
-    return manager;
+namespace {
+
+class ExecutorManagerHolder {
+    std::mutex _mutex;
+    std::shared_ptr<ExecutorManager> _manager = nullptr;
+    int32_t _refCount = 0;
+
+    ExecutorManagerHolder(const ExecutorManagerHolder&) = delete;
+    ExecutorManagerHolder& operator=(const ExecutorManagerHolder&) = delete;
+
+public:
+    ExecutorManagerHolder() = default;
+    ~ExecutorManagerHolder() = default;
+
+    ExecutorManager::Ptr get(bool addRef) {
+        std::lock_guard<std::mutex> lock(_mutex);
+        if (!_manager) {
+            _manager = std::make_shared<ExecutorManagerImpl>();
+            _refCount = 0;
+        }
+        if (addRef) {
```

Comment:
I don't see any other call of this method with `addRef == false`. Shouldn't it always increase the ref count?

Reply:
There are many places to call executorManager(), which is global, so we don't hope to add refcount at each calling point, only for ov::Core construction we add refCount and decreass refCount in ov::Core destruction.